### PR TITLE
Introduce in-profile locking

### PR DESCRIPTION
**Pull request by ia3andy**
- Introduce in-profile locking `-Dlocker.in-profile`
- Change Locker BOM directory to `locker/pom.xml` (previously `.locker/pom.xml)`
- Support for old Locker BOM location
- Allow to switch from `in-profile` mode to `locker-bom` mode
- Update Readme: https://github.com/mvnpm/locker/tree/in-profile
- Add a suite of integration tests

### CODE REVIEW

**Comment by ia3andy** [comment]
@cescoffier would you give it a try before I release?

Have a look to the new readme: https://github.com/mvnpm/locker/tree/in-profile

File: README.md [hunk 1]
```diff
@@ -5,61 +5,69 @@
...
 [![Apache License, Version 2.0, January 2004](https://img.shields.io/github/license/apache/maven.svg?label=License)](https://www.apache.org/licenses/LICENSE-2.0)
 
-The mvnpm locker Maven Plugin will create a version locker BOM for your `org.mvnpm` and `org.webjars` dependencies.
-Allowing you to mimick the `npm-shrinkwrap.json` and `yarn.lock` files in a Maven world.
+The mvnpm locker Maven Plugin will create a version locker profile for your `org.mvnpm` and `org.webjars` dependencies.
+Allowing you to mimick the `package-lock.json` and `yarn.lock` files in a Maven world.
 
-It is essential as NPM dependencies are typically deployed using version ranges, without locking your builds will use different versions of dependencies between builds if any of your transitive NPM based dependencies are updated. 
+_It is essential as NPM dependencies are typically deployed using version ranges, without locking your builds will use different versions of dependencies between builds if any of your transitive NPM based dependencies are updated._
```
Comment (maxandersen) [inline README.md:11]: the cursive rendering doesn't add value does it ? just makes it harder to read?

File: README.md [hunk 2]
```diff
@@ -5,61 +5,69 @@
...
-- a. Checkout the PR locally and use `mvnpm-repo` profile to build. It will make sure all missing transitive dependencies are synchronized on Maven Central.
-- b. Configure your CI to use the `mvnpm-repo` on dependabot updates PRs.
-
-
-## Install the locker BOM in the Maven local repository
+**Now you need to install the locker BOM in the Maven local repository (It is required before building your project):**
 
-It is required before building your project (else it will fail).
+=> On a standalone project:
+- Manually `cd locker && mvn clean install`. Add it to your install doc and add as a new step in your CI.
```
Comment (maxandersen) [inline README.md:37]:
```suggestion
- Manually `mvn -f .locker clean install`. Add it to your install doc and add as a new step in your CI.
```

this works on all OS's

File: README.md (hunk 1 above)
Comment (ia3andy) [reply README.md:11]: In my mind, it was to show that this text is just to add a bit of context. But if you think it's harder to read I can remove it

**Pull request merged**